run-clang-tidy.sh

```diff
@@ -8,23 +8,23 @@ ROOT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
 
 FIX=$1
 
-# We are currently standardized on using LLVM/Clang14 for this script.
+# We are currently standardized on using LLVM/Clang15 for this script.
```

Comment:
We need to actually try clang-tidy-15 and clang-format-15 before we commit to this -- e.g., if the new version changes formatting in an unacceptable way. Let me do some experimenting.

Reply:
OK, the new clang-format looks good (just a few minor changes, all of which are good). clang-tidy-15 doesn't work at all, however :-/

Reply:
OK, let me suggest that you revert the changes to these two scripts (but keep the rest); I'm going to try to land updates to these scripts as a separate PR (since there is some other stuff to update too)